lib/pdf_fill/forms/va210781v2.rb

```diff
@@ -662,17 +666,41 @@ def set_treatment_selection
         @form_data['noTreatment'] = treated ? 0 : 1
       end
 
-      def set_reports_selection
-        reports = @form_data['reports']
-        return if reports.nil?
-
-        @form_data['reportFiled'] = reports['yes'] ? 0 : nil
-        @form_data['noReportFiled'] = reports['no'] ? 1 : nil
-        @form_data['restrictedReport'] = reports['restricted'] ? 0 : nil
-        @form_data['unrestrictedReport'] = reports['unrestricted'] ? 1 : nil
-        @form_data['neitherReport'] = reports['neither'] ? 2 : nil
-        @form_data['policeReport'] = reports['police'] ? 3 : nil
-        @form_data['otherReport'] = reports['other'] ? 4 : nil
+      def process_reports
+        report_filed = false
+        no_report = false
+        police_reports = []
+        other_reports = []
+        reports_details = @form_data['reportsDetails'] ||= {}
+
+        @form_data['events'].each do |event|
+          reports = event['reports'] || {}
+          other_report = event['otherReports']
+          next if reports.empty? && other_report&.blank?
+
+          report_filed ||= reports.except('none').values.include?(true)
+          no_report ||= reports['none']
+
+          set_report_types(other_report, report_filed, reports)
+
+          police_report = format_police_details(event)
+          police_reports << police_report unless police_report.empty?
+
+          other_reports << other_report if other_report.present?
+        end
+
+        reports_details['police'] = police_reports.join('; ') unless police_reports.empty?
+        reports_details['other'] = other_reports.join('; ') unless other_reports.empty?
+        @form_data['noReportFiled'] = no_report && !report_filed ? 1 : nil
+      end
+
+      def set_report_types(other_report, report_filed, reports)
+        @form_data['reportFiled'] ||= report_filed ? 0 : nil
+        @form_data['restrictedReport'] ||= reports['restricted'] ? 0 : nil
+        @form_data['unrestrictedReport'] ||= reports['unrestricted'] ? 1 : nil
+        @form_data['neitherReport'] ||= reports['neither'] ? 2 : nil
+        @form_data['policeReport'] ||= reports['police'] ? 3 : nil
+        @form_data['otherReport'] ||= other_report ? 4 : nil
```

Comment:
Am I supposed to understand what these number mean? 😟 😅 If it would be helpful to future teams working on this, can you leave a comment here?

Reply:
Numeric values correspond to specific checkbox selections on the PDF form:
- 0: Indicates 'reportFiled' or 'restrictedReport.'
- 1: Indicates 'unrestrictedReport.'
- 2: Indicates 'neitherReport.'
- 3: Indicates 'policeReport.'
- 4: Indicates 'otherReport.'
      
If the corresponding condition is true, the value is set and the checkbox gets marked; otherwise, it remains nil.

Reply:
@rmtolmach Added in [89e1e5f](https://github.com/department-of-veterans-affairs/vets-api/pull/20663/commits/89e1e5f348c47cd04402167889da484a4f22fb0f)